go/types/map_editor.go

```diff
@@ -38,14 +47,56 @@ func (me *MapEditor) Build(vrw ValueReadWriter) Map {
 
 	me.normalize()
 
-	var ch *sequenceChunker
-	for i, kv := range me.edits {
-		if i+1 < len(me.edits) && me.edits[i+1].key.Equals(kv.key) {
-			continue // next edit supercedes this one
+	cursChan := make(chan chan *sequenceCursor)
+	kvsChan := make(chan chan mapEntry)
+
+	go func() {
+		for i, edit := range me.edits {
+			if i+1 < len(me.edits) && me.edits[i+1].key.Equals(edit.key) {
+				continue // next edit supercedes this one
+			}
+
+			// TODO: Use ReadMany
+			cc := make(chan *sequenceCursor, 1)
+			cursChan <- cc
+
+			go func(k Value, cc chan *sequenceCursor) {
+				cc <- newCursorAtValue(me.m.seq, k, true, false, false)
+			}(edit.key, cc)
```

Comment:
There is no need to pass `cc` -> it's already local to each time around the loop. And wrt `edit.key`, is more idiomatic in Go to reassign to a local:

```
for .... {
  edit := edit
  ...
}
```